azerothcore/azerothcore-wotlk
# fix(Docker): Remove db-import container, automatically import module SQL

Remove db-import container and supporting files since worldserver and authserver now do database imports automatically. Enable automatic imports of module SQL files.



## Changes Proposed:
- docker-compose.yml: remove db-import container and remove all references in other services
- docker-compose.yml: add volume mapping for worldserver to modules to allow for SQL imports to happen automatically
- docker-compose.yml: change binaries volume on worldserver and authserver to be read/write, since the DB import process [creates a temporary file in the directory](https://github.com/kjack9/azerothcore-wotlk/blob/0bf649a3f5ff131876b6d27f34c287b8e3a512d3/src/server/database/Updater/DBUpdater.cpp#L189) if a database needs to be created
- apps/docker/Dockerfile: remove references to dbimport configuration files
- env/docker/etc/dbimport.conf.dockerdist: delete
- env.ac: add clarifying comments
- env.docker: add clarifying comments

## Issues Addressed:

- None

## SOURCE:

Personal experience while setting up a family server. After troubleshooting module SQL importing with `mynameismeat` on Discord, we noticed some improvements that could be made to eliminate the db-import container entirely and re-enable module SQL imports. This PR is the result.

## Tests Performed:

- server built from scratch using this modifications
- added a module, did a rebuild, and observed the SQL being properly imported


## How to Test the Changes:


1. Follow standard docker deployment instructions for AzerothCore
2. Observe that on first run that the database is successfully created by authserver and worldserver
3. Add a module, rebuild, and restart the server. Observe that worldserver successfully imports the sql file.

## Known Issues and TODO List:

- module sql files are only imported if in the `mod_name/data/sql/db-<db_name>/` directory, a format which many modules do not adhere to. `mynameismeat` intends to correct this in their own PR.
- if [this line](https://github.com/kjack9/azerothcore-wotlk/blob/0bf649a3f5ff131876b6d27f34c287b8e3a512d3/src/server/database/Updater/DBUpdater.cpp#L189) was changed to point to a temporary folder, the binaries volume could be switched back to read-only.



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
